include/frozen/bits/algorithms.h

```diff
@@ -69,6 +69,18 @@ constexpr auto min_element(Iter begin, const Iter &end,
   return result;
 }
 
+template <class T> constexpr void cswap(T &a, T &b) {
+  auto tmp = a;
+  a = b;
+  b = tmp;
+}
+
+template <class T0, class T1>
+constexpr void cswap(std::tuple<T0, T1> &a, std::tuple<T0, T1> &b) {
+  cswap(std::get<0>(a), std::get<0>(b));
+  cswap(std::get<1>(a), std::get<1>(b));
+}
+
```

Comment:
I was very close to theses solutions too.
Just the time to remember the `operator ,`  trick ;-)